.circleci/config.yml

```diff
@@ -0,0 +1,103 @@
+version: 2
+
+defaults: &defaults
+  working_directory: ~/project/yarn
+  docker:
+    - image: yarnpkg/dev:latest
+
+jobs:
+  install:
+    <<: *defaults
+    steps:
+      - checkout
+      - restore_cache:
+          key: yarn-{{ .Branch }}-{{ checksum "yarn.lock" }}
+      - run:
+          name: Install Dependencies
+          command: yarn install
+      - save_cache:
+          key: yarn-{{ .Branch }}-{{ checksum "yarn.lock" }}
+          paths:
+            - .cache/yarn
+      - persist_to_workspace:
+          root: ~/project
+          paths:
+            - yarn
+  test:
+    <<: *defaults
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/project
+      - run:
+          name: Tests
+          command: |
+            node -v
+            if [ "$CIRCLE_BRANCH" == 'master' ]; then
+              ./scripts/set-dev-version.js
+            fi;
+            yarn lint
+            # Limit maxWorkers to 3 to avoid OOM on CircleCI
+            yarn test-ci -- -- --maxWorkers 3
+            yarn check-lockfile
+  build:
+    <<: *defaults
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/project
+      - run:
+          name: Build distribution
+          command: |
+            node -v
+            if [ "$CIRCLE_BRANCH" == 'master' ]; then
+              ./scripts/set-dev-version.js
+            fi;
+            yarn build-dist
+            ./scripts/build-deb.sh
+      - store_artifacts:
+          path: artifacts/
+          destination: yarnpkg
+      - persist_to_workspace:
+          root: ~/project
+          paths:
+            - yarn
+  deploy:
+    <<: *defaults
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/project
+      - run:
+          name: Deploy
+          command: |
+            # Only NPM is handled here - All other release files are handled in a webhook.
+            if [ "${CIRCLE_PROJECT_USERNAME}" == "yarnpkg" ]; then
+              if git describe --tags --abbrev=0 `git rev-parse HEAD` | grep "^v[0-9]\+\.[0-9]\+\.[0-9]\+$"; then
```

Comment:
Could this use `CIRCLE_TAG` instead? https://circleci.com/docs/2.0/env-vars/#build-details

Reply:
Oops, checking. Thanks.

Reply:
I do not know why, but `CIRCLE_TAG` is always empty.(like this https://discuss.circleci.com/t/build-on-tag/9864/43) Instead, I've just tried git tag execution but filtering did not work as we expected.

When I have more time I would like to settle down to study, but regarding this pull request, I want to proceed as it is.

Reply:
In CircleCI 2.0, this is done using the `filter` prop of the workflow jobs.

You could do something like this:

```yml
default_filters: &default_filters
  tags:
    only: '/v[0-9]+(\.[0-9]+)*/'


# …

workflows:
  version: 2
  install-test-build-and-deploy:
    jobs:
      - install:
          filters: *default_filters
      - test:
          filters: *default_filters
          requires:
            - install
      - build:
          filters: *default_filters
          requires:
            - install
      - deploy:
          filters:
            << *default_filters
            branches:
              ignore: '/.*/'
          requires:
            - test
            - build
```

Branches are opt-out and tags are opt-in. So this will run jobs for all tags matching the regexp, but only the `install`, `test` and `build` jobs will be ran for all branches.

Reply:
thanks! I will try it.

Reply:
@tusbar CircleCI tag filter is not recognized properly. I would be happy if you could tell me the specific steps of git tag release.

Reply:
@serima I’m using it successfully here: https://github.com/etalab/inspire/blob/master/.circleci/config.yml


- Here’s a successful tag workflow: https://circleci.com/workflow-run/f0b8eb73-e846-4727-b0ff-1ed724d1375f
- And a branch: https://circleci.com/workflow-run/f608ea8f-5c3e-4631-9b94-07d01bee2bf2

Reply:
@tusbar cool. thanks 🤓 I will try it again.